libraries/cbor/src/macros.rs

```diff
@@ -41,20 +41,31 @@ macro_rules! cbor_map_options {
     };
 
     ( $( $key:expr => $value:expr ),* ) => {
+        {
+            let mut _map = ::alloc::collections::BTreeMap::<_, $crate::values::Value>::new();
+            extend_cbor_map_options! ( &mut _map, $( $key => $value, )* );
+            $crate::values::Value::Map(_map)
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! extend_cbor_map_options {
+    // Add trailing comma if missing.
+    ( &mut $initial:expr, $( $key:expr => $value:expr ),+ ) => {
+        extend_cbor_map_options! ( &mut $initial, $($key => $value,)+ )
+    };
+
+    ( &mut $initial:expr, $( $key:expr => $value:expr, )* ) => {
         {
             // The import is unused if the list is empty.
             #[allow(unused_imports)]
             use $crate::values::{IntoCborKey, IntoCborValueOption};
-            let mut _map = ::alloc::collections::BTreeMap::<_, $crate::values::Value>::new();
             $(
-            {
-                let opt: Option<$crate::values::Value> = $value.into_cbor_value_option();
```

Comment:
IIRC it was necessary to write the type here so that "into" knows into what type to convert it. In any case, the explicit type doesn't hurt readability of the macro's code.

Reply:
It might have been in the past, but now at least it's not. The return type is specified by `into_cbor_value_option`.